host_core/lib/host_core/jetstream/legacy_cache_loader.ex

```diff
@@ -0,0 +1,92 @@
+defmodule HostCore.Jetstream.LegacyCacheLoader do
+  @moduledoc """
+  The legacy cache loader is responsible for loading metadata from the deprecated LATTICECACHE_{prefix} stream
+  into the in-memory caches for claims, oci refs, call aliases, and linkdefs.
+  """
+
+  alias Phoenix.PubSub
+
+  require Logger
+  #  use Gnat.Server
+
+  import HostCore.ControlInterface.LatticeServer, only: [failure_ack: 1]
+
+  def handle_legacy_request(js_domain, topic, body) do
+    case String.split(topic, ".", parts: 3) do
+      [_lc, prefix, remainder] ->
+        remainder
+        |> String.split(".")
+        |> List.to_tuple()
+        |> handle_request(body, prefix, js_domain)
+
+      _ ->
+        {:reply, failure_ack("Invalid request topic")}
+    end
+
+    {:reply, ""}
+  end
+
+  def handle_request({"linkdefs", key}, body, prefix, js_domain) do
+    case Jason.decode(body, keys: :atoms) do
+      {:ok, %{deleted: true} = ld} ->
+        HostCore.Linkdefs.Manager.uncache_link_definition(
+          prefix,
+          ld.id
+        )
+
+        Logger.debug(
+          "Skipping migration of deleted linkdef from #{ld.actor_id} to #{ld.provider_id}"
+        )
+
+      {:ok, ld} ->
+        HostCore.Jetstream.Client.kv_put(prefix, js_domain, "LINKDEF_#{ld.id}", ld)
```

Comment:
it will make its way to the ETS table because the (non legacy) cache monitor is listening to changes on the bucket and then pushing the data into the ets table. If we wrote to the table here we'd be double-writing.